src/cli/config-flags.ts

```diff
@@ -0,0 +1,137 @@
+import type { LogLevel, TaskCommand } from '@stencil/core/declarations';
+
+/**
+ * All the Boolean options supported by the Stencil CLI
+ */
+export const BOOLEAN_CLI_ARGS = [
+  'build',
+  'cache',
+  'checkVersion',
+  'ci',
+  'compare',
+  'debug',
+  'dev',
+  'devtools',
+  'docs',
+  'e2e',
+  'es5',
+  'esm',
+  'headless',
+  'help',
+  'log',
+  'open',
+  'prerender',
+  'prerenderExternal',
+  'prod',
+  'profile',
+  'serviceWorker',
+  'screenshot',
+  'serve',
+  'skipNodeCheck',
+  'spec',
+  'ssr',
+  'stats',
+  'updateScreenshot',
+  'verbose',
+  'version',
+  'watch',
+] as const;
+
+/**
+ * All the Number options supported by the Stencil CLI
+ */
+export const NUMBER_CLI_ARGS = ['maxWorkers', 'port'] as const;
+
+/**
+ * All the String options supported by the Stencil CLI
+ */
+export const STRING_CLI_ARGS = ['address', 'config', 'docsJson', 'emulate', 'root', 'screenshotConnector'] as const;
+
+/**
+ * All the LogLevel-type options supported by the Stencil CLI
+ *
+ * This is a bit silly since there's only one such argument atm,
+ * but this approach lets us make sure that we're handling all
+ * our arguments in a type-safe way.
+ */
+export const LOG_LEVEL_CLI_ARGS = ['logLevel'] as const;
```

Comment:
this is obvs pretty similar to a string arg, but because we parse it into a `LogLevel` union type which is `"info" | "warn" | "debug" | "error"` we need to handle it separately